net-p2p/bitcoin-qt/bitcoin-qt-0.21.0.ebuild

```diff
@@ -0,0 +1,187 @@
+	>=dev-libs/univalue-1.0.4:=
+	dev-qt/qtcore:5
+	dev-qt/qtgui:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtwidgets:5
+	system-leveldb? ( virtual/bitcoin-leveldb )
+	dbus? ( dev-qt/qtdbus:5 )
+	dev-libs/libevent:=
+	qrcode? (
+		media-gfx/qrencode:=
+	)
+	sqlite? ( >=dev-db/sqlite-3.7.17:= )
+	upnp? ( >=net-libs/miniupnpc-1.9.20150916:= )
+	wallet? ( sys-libs/db:$(db_ver_to_slot "${DB_VER}")=[cxx] )
+	zeromq? ( net-libs/zeromq:= )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	>=sys-devel/autoconf-2.69
+	>=sys-devel/automake-1.13
+	dev-qt/linguist-tools:5
+	knots? (
+		gnome-base/librsvg
+		media-gfx/imagemagick[png]
+	)
+"
+
+DOCS=(
+	doc/bips.md
+	doc/bitcoin-conf.md
+	doc/descriptors.md
+	doc/files.md
+	doc/JSON-RPC-interface.md
+	doc/psbt.md
+	doc/reduce-traffic.md
+	doc/release-notes.md
+	doc/REST-interface.md
+	doc/tor.md
+)
+
+S="${WORKDIR}/bitcoin-${BITCOINCORE_COMMITHASH}"
+
+pkg_pretend() {
+	if use knots; then
+		elog "You are building ${PN} from Bitcoin Knots."
+		elog "For more information, see:"
+		elog "https://bitcoinknots.org/files/0.21.x/${KNOTS_PV}/${KNOTS_P}.desc.html"
+	else
+		elog "You are building ${PN} from Bitcoin Core."
+		elog "For more information, see:"
+		elog "https://bitcoincore.org/en/2021/01/14/release-${PV}/"
+	fi
+	elog "Replace By Fee policy is now always enabled by default: Your node will"
+	elog "preferentially mine and relay transactions paying the highest fee, regardless"
+	elog "of receive order. To disable RBF, set mempoolreplacement=never in bitcoin.conf"
+}
+
+src_prepare() {
+	sed -i 's/^\(complete -F _bitcoind \)bitcoind \(bitcoin-qt\)$/\1\2/' contrib/bitcoind.bash-completion || die
+
+	# Save the generic icon for later
+	cp src/qt/res/src/bitcoin.svg bitcoin128.svg || die
+
+	local knots_patchdir="${WORKDIR}/${KNOTS_P}.patches/"
+
+	eapply "${knots_patchdir}/${KNOTS_P}.syslibs.patch"
+
+	if use knots; then
+		eapply "${knots_patchdir}/${KNOTS_P}.f.patch"
+		eapply "${knots_patchdir}/${KNOTS_P}.branding.patch"
+		eapply "${knots_patchdir}/${KNOTS_P}.ts.patch"
+	fi
+
+	eapply_user
+
+	echo '#!/bin/true' >share/genbuild.sh || die
+	mkdir -p src/obj || die
+	echo "#define BUILD_SUFFIX gentoo${PVR#${PV}}" >src/obj/build.h || die
+
+	eautoreconf
+	rm -r src/secp256k1 || die
+	if use system-leveldb; then
+		rm -r src/leveldb || die
+	fi
+}
+
+src_configure() {
+	local my_econf=(
+		$(use_enable asm)
+		$(use_with dbus qtdbus)
+		$(use_with qrcode qrencode)
+		$(use_with upnp miniupnpc)
+		$(use_enable upnp upnp-default)
+		$(use_enable test tests)
+		$(use_enable wallet)
+		$(use_enable zeromq zmq)
+		--with-gui=qt5
+		--disable-util-cli
+		--disable-util-tx
+		--disable-util-wallet
+		--disable-bench
+		--without-libs
+		--without-daemon
+		--disable-fuzz
+		--disable-ccache
+		--disable-static
+		$(use_with sqlite)
+		$(use_with system-leveldb)
+		--with-system-libsecp256k1
+		--with-system-univalue
+	)
+	econf "${my_econf[@]}"
+}
+
+src_install() {
+	default
+
+	rm -f "${ED}/usr/bin/test_bitcoin" || die
```

Comment:
We generally don't use `rm -f` because they don't error out if the file is missing, so these lines will be left to ebuilds for eternity.